### PR TITLE
BAH-940: Ability to provide custom properties

### PR DESCRIPTION
**Pull request by chinzou**
https://bahmni.atlassian.net/browse/BAH-940

### CODE REVIEW

File: openelis/src/org/bahmni/feed/openelis/AtomFeedProperties.java [hunk 1]
```diff
@@ -18,29 +18,36 @@
...
 public class AtomFeedProperties {
-    
+
     private static final String FEED_CONNECT_TIMEOUT = "feed.connectionTimeoutInMilliseconds";
     private static final String FEED_REPLY_TIMEOUT = "feed.replyTimeoutInMilliseconds";
     private static final String FEED_MAX_FAILED_EVENTS = "feed.maxFailedEvents";
     private static final String FAILED_EVENT_MAX_RETRY = "feed.failedEventMaxRetry";
 
     public static final String DEFAULT_PROPERTY_FILENAME = "/atomfeed.properties";
+    public static final String CUSTOM_PROPERTY_FILENAME = System.getProperty("atomfeed.properties.file");
```
Comment (rbuisson) [inline openelis/src/org/bahmni/feed/openelis/AtomFeedProperties.java]: @chinzou , could you rename this property to be:
`public static final String CUSTOM_PROPERTY_FILENAME = System.getProperty("ATOMFEED_PROPERTIES_FILE");`

File: openelis/src/us/mn/state/health/lims/hibernate/HibernateUtil.java [hunk 2]
```diff
@@ -45,6 +49,7 @@
     private static final ThreadLocal threadSession = new ThreadLocal();
     private static final ThreadLocal threadInterceptor = new ThreadLocal();
     private static String CONFIG_FILE_LOCATION = "/us/mn/state/health/lims/hibernate/hibernate.cfg.xml";
+    private static String CUSTOM_FILE_PROPERTY = System.getProperty("hibernate.properties.file");
```
Comment (rbuisson) [inline openelis/src/us/mn/state/health/lims/hibernate/HibernateUtil.java]: Same thing here, could you rename to:
`    private static String CUSTOM_FILE_PROPERTY = System.getProperty("HIBERNATE_PROPERTIES_FILE");
`

File: openelis/src/org/bahmni/feed/openelis/AtomFeedProperties.java [hunk 3]
```diff
@@ -18,29 +18,36 @@
...
     private AtomFeedProperties() {
         InputStream propertyStream = null;
         try {
             propertyStream = this.getClass().getResourceAsStream(DEFAULT_PROPERTY_FILENAME);
             properties = new Properties();
             properties.load(propertyStream);
 
+            if ((new File(CUSTOM_PROPERTY_FILENAME)).exists()){
+                FileInputStream customPropertyStream = new FileInputStream(CUSTOM_PROPERTY_FILENAME);
+                properties.load(customPropertyStream);
```
Comment (angshu) [inline openelis/src/org/bahmni/feed/openelis/AtomFeedProperties.java]: suppose say. .. the CUSTOM_PROPERTY_FILENAME has only 3 properties. do all the properties get replaced, or only the 3?

File: openelis/src/us/mn/state/health/lims/hibernate/HibernateUtil.java [hunk 4]
```diff
@@ -59,9 +64,18 @@
     static {
         try {
             configuration = new Configuration();
+            configuration.configure(configFile);
+
+            if ((new File(CUSTOM_FILE_PROPERTY)).exists()){
+                FileInputStream customPropertyStream = new FileInputStream(CUSTOM_FILE_PROPERTY);
+                Properties properties = new Properties();
+                properties.load(customPropertyStream);
```
Comment (angshu) [inline openelis/src/us/mn/state/health/lims/hibernate/HibernateUtil.java]: same question as above.

File: openelis/src/org/bahmni/feed/openelis/AtomFeedProperties.java (hunk 3 above)
Comment (angshu) [reply openelis/src/org/bahmni/feed/openelis/AtomFeedProperties.java]: because in that case, we ought to do a merge

Comment (chinzou) [reply openelis/src/org/bahmni/feed/openelis/AtomFeedProperties.java]: if the CUSTOM_PROPERTY_FILENAME has 3 properties, only those 3 properties get replaced

File: openelis/src/us/mn/state/health/lims/hibernate/HibernateUtil.java (hunk 4 above)
Comment (chinzou) [reply openelis/src/us/mn/state/health/lims/hibernate/HibernateUtil.java]: yes, it's the same behavior. only the properties in the custom file are overridden

**Pull request merged**